pkg/trace/agent/service.go

```diff
@@ -119,11 +119,11 @@ const appType = "app_type"
 
 // TraceServiceExtractor extracts service metadata from top-level spans
 type TraceServiceExtractor struct {
-	outServices chan<- pb.ServicesMetadata
+	outServices chan pb.ServicesMetadata
```

Comment:
I'm probably missing something, but why did we need to change the channel type?

Reply:
Oh nevermind I see that we're reading from this channel in one of the benchmark tests

Reply:
It's definitely not nice to change this type. But I assumed it's a worthwhile compromise given that it allows us to properly benchmark the `Process` function in order to improve the performance and clean up the code. I think the end result is definitely worth it. Eventually we should add the type back here.